src/charm.py

```diff
@@ -160,22 +161,21 @@ def _configure_kubeproxy(self, event):
             external_cloud_provider=self.external_cloud_provider,
         )
 
-    def _configure_labels(self):
-        """Configure labels."""
-        if not os.path.exists("/root/.kube/config"):
-            log.info("Waiting for kubeconfig before configuring labels")
-            return
-
-        status.add(ops.MaintenanceStatus("Configuring node labels"))
-
+    @status.on_error(ops.WaitingStatus("Waiting to apply node labels"))
+    def _apply_node_labels(self):
+        """Apply node labels."""
+        status.add(ops.MaintenanceStatus("Apply Node Labels"))
         if self.label_maker.active_labels() is not None:
             self.label_maker.apply_node_labels()
+            log.info("Node %s labelled successfully", self.get_node_name())
+        else:
+            raise status.ReconcilerError("Failed to apply node labels")
```

Comment:
@HomayoonAlimohammadi Hows this looking?  I even added an integration test for it

Reply:
Awesome! Thanks a lot!